packages/core-app-api/alpha-api-report.md

```diff
@@ -5,32 +5,18 @@
 ```ts
 import { AppTranslationApi } from '@backstage/core-plugin-api/alpha';
 import { i18n } from 'i18next';
+import { TranslationMessages } from '@backstage/core-plugin-api/alpha';
 import { TranslationRef } from '@backstage/core-plugin-api/alpha';
+import { TranslationResource } from '@backstage/core-plugin-api/alpha';
 
 // @alpha (undocumented)
 export class AppTranslationApiImpl implements AppTranslationApi {
   // (undocumented)
-  addLazyResources<Messages extends Record<string, string>>(
-    translationRef: TranslationRef<Messages>,
-    initResources?: Record<
-      string,
-      () => Promise<{
-        messages: TranslationMessages<TranslationRef>;
-      }>
-    >,
-  ): void;
+  addLazyResources(resource: TranslationResource): void;
   // (undocumented)
-  addResources<Messages extends Record<string, string>>(
-    translationRef: TranslationRef<Messages>,
-    initResources?: Record<
-      string,
-      TranslationMessages<TranslationRef<Messages>>
-    >,
-  ): void;
+  addMessages(messages: TranslationMessages): void;
   // (undocumented)
-  addResourcesByRef<Messages extends Record<string, string>>(
-    translationRef: TranslationRef<Messages>,
-  ): void;
+  addResource(translationRef: TranslationRef): void;
```

Comment:
tempting to just make this take `TranslationRef | TranslationResource` ... of hey, a single method taking `TranslationRef | TranslationResource | TranslationMessages`. From a consumer standpoint it doesn't provide much value to have these as three different methods, and one being about lazy stuff isn't really valuable to discern for, from the user's POV

Reply:
Plan is to make this a `getX(ref: TranslationRef)` and not have the resources or messages be referenced directly here at all. Where X is something that lets you generate translations :p